gammapy/modeling/prior.py

```diff
@@ -0,0 +1,111 @@
+from . import Parameter
+
+
+class Prior:
+    """
+    Base prior class containing methods to serialise.
+    """
+
+    def __call__(self, base):
+        if isinstance(base, Parameter):
+            return self.evaluate(base.value)
+        else:  # model-like instance
+            return self.evaluate(base.parameters.value)
+
+    def to_dict(self):
+        output = {
+            "tag": self.tag[0],
+        }
+        output["parameters"] = [
+            {"name": p[0], "value": p[1]} for p in self.prior_parameters
+        ]
+        return output
+
+    def __str__(self):
+        string = f"{self.__class__.__name__}\n"
+        string += "-" * 10
+        for p in self.prior_parameters:
+            string += f"\n{p[0]:10}: {p[1]}"
+        return string
+
+
+class GaussianPrior(Prior):
+    """Gaussian Prior with mu and sigma."""
+
+    tag = ["GaussianPrior"]
+
+    def __init__(self, mu=0, sigma=1):
+        self.mu = mu
+        self.sigma = sigma
+
+    def evaluate(self, value):
+        return ((value - self.mu) / self.sigma) ** 2
+
+    @property
+    def prior_parameters(self):
+        return [("mu", self.mu), ("simga", self.sigma)]
```

Comment:
I am wondering whether one should not add a "StepPrior", ie 0 for X<Xmin and X>Xmax (Xmin and Xmax could be infinite) and 1 else where...
This can be nice to e.g. force a flux to be positive.